bin/main.rs

```diff
@@ -1,3 +1,372 @@
+            description: Some("A stream that stores all commands for wadm".to_string()),
+            num_replicas: 1,
+            retention: async_nats::jetstream::stream::RetentionPolicy::WorkQueue,
+            subjects: vec![DEFAULT_COMMANDS_TOPIC.to_owned()],
+            max_age: DEFAULT_EXPIRY_TIME,
+            storage: async_nats::jetstream::stream::StorageType::File,
+            allow_rollup: false,
+            ..Default::default()
+        })
+        .await
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+
+    let host_id = args
+        .host_id
+        .unwrap_or_else(|| uuid::Uuid::new_v4().to_string());
+    tokio::select! {
+        res = handle_events(event_stream, context.clone(), host_id) => {
+            res?
+        }
+        res = handle_commands(command_stream) => {
+            res?
+        }
+    }
+    Ok(())
+}
+
+// NOTE: These functions will probably need to be replaced with something in the library code, but
+// they're good enough for testing
+async fn handle_events(stream: Stream, context: Context, host_id: String) -> anyhow::Result<()> {
+    let mut consumer = EventConsumer::new(stream, "wasmbus.evt.default")
+        .await
+        .map_err(|e| anyhow::anyhow!("{e:?}"))?;
+    while let Some(res) = consumer.next().await {
+        match res {
+            Err(e) => {
+                error!(error = %e, "Error when trying to fetch message from consumer");
+                continue;
+            }
+            Ok(mut evt) => {
+                debug!(event = ?evt.as_ref(), "Handling received event");
+                // THIS IS WHERE WE'D DO REAL WORK
+                if let Err(e) = evt.ack().await {
+                    warn!(error = %e, "Error when acking");
+                    // If we can't ack, loop back around to try on the redelivery
+                    continue;
+                }
+
+                // NOTE: There is a possible race condition here where we send the command, and it
+                // doesn't work even though we've acked the message. Worst case here is that we end
+                // up not starting/creating something, which would be fixed on the next heartbeat.
+                // This is better than the other option of double starting something when the ack
+                // fails (think if something resulted in starting 100 actors on a host and then it
+                // did it again)
+                if let Err(e) = send_fake_command(&context, &host_id, &evt).await {
+                    error!(error = %e, "Got error when sending command, will have to catch up on next reconcile");
+                }
+            }
+        }
+    }
+    Ok(())
+}
+
+// For testing only, to send a fake command in response to an event
+async fn send_fake_command(context: &Context, host_id: &str, event: &Event) -> anyhow::Result<()> {
+    use wadm::nats_utils::ensure_send;
+    let command: Command = match event {
+        Event::ActorStarted(actor) => {
```

Comment:
I know this is test, so I'm assuming this is just mock data... but wadm wouldn't emit a start actor command in response to an actor started event.

Reply:
Yep, totally something to demo it works and so I knew what would be coming out the other end!